src/core/jit/request.rs

```diff
@@ -6,28 +6,43 @@ use serde::Deserialize;
 use super::{Builder, Error, ExecutionPlan, Result};
 use crate::core::blueprint::Blueprint;
 
-#[derive(Debug, Deserialize, Setters)]
+#[derive(Debug, Deserialize, Setters, Clone)]
 pub struct Request<Value> {
     #[serde(default)]
     pub query: String,
     #[serde(default, rename = "operationName")]
     pub operation_name: Option<String>,
     #[serde(default)]
-    pub variables: HashMap<String, Value>,
+    pub variables: Variables<Value>,
     #[serde(default)]
     pub extensions: HashMap<String, Value>,
 }
 
+#[derive(Default, Debug, Deserialize, Clone)]
+pub struct Variables<Value>(HashMap<String, Value>);
+
+impl<Value> Variables<Value> {
+    pub fn new() -> Self {
+        Self(HashMap::new())
+    }
+    pub fn get(&self, key: &str) -> Option<&Value> {
+        self.0.get(key)
+    }
+    pub fn insert(&mut self, key: String, value: Value) {
+        self.0.insert(key, value);
+    }
+}
+
 impl From<async_graphql::Request> for Request<async_graphql_value::ConstValue> {
     fn from(value: async_graphql::Request) -> Self {
         Self {
             query: value.query,
             operation_name: value.operation_name,
             variables: match value.variables.into_value() {
-                async_graphql_value::ConstValue::Object(val) => {
-                    HashMap::from_iter(val.into_iter().map(|(k, v)| (k.to_string(), v)))
-                }
-                _ => HashMap::new(),
+                async_graphql_value::ConstValue::Object(val) => Variables(HashMap::from_iter(
+                    val.into_iter().map(|(k, v)| (k.to_string(), v)),
+                )),
```

Comment:
Should have helpful constructors to create Variables from iters so that we don't need to write all this here.